src/identities/mnemonic.ts

```diff
@@ -133,10 +143,22 @@ export default class Mnemonic extends Identity implements IdentityInterface {
     }).confidentialAddress!;
   }
 
-  private getAddress(isChange: boolean): AddressInterface {
-    const currentIndex = this.index;
+  // store the generation inside local cache
+  private persistAddressToCache(address: AddressInterfaceExtended): void {
+    const privateKeyBuffer = Buffer.from(address.signingPrivateKey, 'hex');
+    const publicKey: Buffer = ECPair.fromPrivateKey(privateKeyBuffer, {
+      network: this.network,
+    }).publicKey!;
+    const script = this.scriptFromPublicKey(publicKey);
+    this.scriptToAddressCache.set(script, address);
+  }
+
+  private getAddress(
+    isChange: boolean,
+    index: number = this.index
```

Comment:
since you are forcing as default here the external chain index, you should force the isChange as well

Reply:
I just remove the default parameter (I do not use it)

Reply:
very good article thank you